### PR TITLE
[Addon] Correction of anomaly and improvement - service.xbmc.versioncheck

### PR DESCRIPTION
**Pull request by Cyr-ius**
Correction of anomaly and improvement of the service

## Description
I propose three evolutions of the service
Test if a password is required for sudo
Improved tracks for better visibility
Correct the update of the APT system
Fixed the display in UTF8. It was missing the library import
Replacing message boxes with dialog background

A commit by update

v0.3.30
- Replace the message box with a dialog backgroun box
- Choice System update or Kodi update or the both
- Add choice for autoremove old packages if system update
- Fix unicodedata (missing import)

v0.3.281
- Fix Apt Upgrade

v0.3.28
-  Improvement of log's traces

v0.3.27
- Test mandatory password for sudo

## Motivation and Context
The system update was only effective if there was an update of Kodi. I wanted to be able to treat both without depending on one or the other
Fix the display, forget the import of a library
Allow treatment in the background

## How Has This Been Tested?
Test on Ubuntu 17.10, 18.04, Raspberry pi 2 Stretch
with Kodi 18 alpha 2

## Types of change

- [ X ] Bug fix (non-breaking change which fixes an issue)
- [ X ] Improvement (non-breaking change which improves existing functionality)
- [ X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ X  ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ X ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ X ] All new and existing tests passed

### CODE REVIEW

File: addon.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <addon id="service.xbmc.versioncheck"
     name="Version Check"
-    version="0.3.281"
+    version="0.3.30"
```
Comment (Razzeee) [inline addon.xml]: I'm wondering if there's a reason for the high patch number we had here before

File: resources/language/English/strings.po [hunk 2]
```diff
@@ -111,3 +143,11 @@ msgstr ""
 msgctxt "#32035"
 msgid "It is recommended that you to upgrade to a newer version."
 msgstr ""
+
+msgctxt "#32037"
+msgid "Do you want checking updates?"
```
Comment (Razzeee) [inline resources/language/English/strings.po]: This doesn't seem to be used

File: service.py [hunk 3]
```diff
@@ -31,82 +31,99 @@
...
-    from lib.versions import compare_version
-    # retrieve versionlists from supplied version file
-    versionlist = get_versionfilelist()
-    # retrieve version installed
-    version_installed = get_installedversion()
-    # compare installed and available
-    oldversion, version_installed, version_available, version_stable = compare_version(version_installed, versionlist)
-    return oldversion, version_installed, version_available, version_stable
+            self._versionchecklinux(packages)
+        elif ADDON.getSetting("versioncheck_enable") == "true":
```
Comment (Razzeee) [inline service.py]: Seems a bit late to check this? So linux would always be checked, even if you disable this setting?

File: addon.xml (hunk 1 above)
Comment (Cyr-ius) [reply addon.xml]: I proposed 4 commits and in each I propose to add +0.1
but I can suggest a 0.3.27

File: service.py (hunk 3 above)
Comment (Cyr-ius) [reply service.py]: In the default version, Linux does not update if there is no new version of Kodi.
The idea is proposed an update of Kodi, or Linux or both elements.
Knowing that on Linux, I am well aware that he will also update Kodi

File: resources/language/English/strings.po (hunk 2 above)
Comment (Cyr-ius) [reply resources/language/English/strings.po]: J'ai supprimé le numéro inutile et renuméroté le formulaire (view new commit)

Comment (Cyr-ius) [reply resources/language/English/strings.po]: In English
I deleted the useless number and renumbered the form

File: resources/language/English/strings.po [hunk 4]
```diff
@@ -24,7 +24,15 @@ msgctxt "#32002"
 msgid "Visit Kodi.tv for more information."
 msgstr ""
 
-#empty strings from id 32003 to 32008
+#empty strings from id 32003 to 32006
+
+msgctxt "#32007"
+msgid "Check kodi upgrade available."
```
Comment (MartijnKaijser) [inline resources/language/English/strings.po]: Check if Kodi

File: resources/language/English/strings.po [hunk 5]
```diff
@@ -24,7 +24,15 @@ msgctxt "#32002"
...
+msgstr ""
+
+msgctxt "#32008"
+msgid "Check system upgrade available."
```
Comment (MartijnKaijser) [inline resources/language/English/strings.po]: Check if

File: resources/language/English/strings.po [hunk 6]
```diff
@@ -58,7 +66,17 @@ msgctxt "#32016"
 msgid "There is a newer stable Kodi version available."
 msgstr ""
 
-#empty strings from id 32017 to 32019
+msgctxt "#32017"
+msgid "Do you want to system restart to finish the upgrade?"
```
Comment (MartijnKaijser) [inline resources/language/English/strings.po]: restart the system

File: resources/language/English/strings.po [hunk 7]
```diff
@@ -80,7 +98,21 @@ msgctxt "#32024"
 msgid "Linux: Upgrade using apt"
 msgstr ""
 
-#empty strings from id 32025 to 32029
+msgctxt "#32025"
+msgid "Use sudo"
+msgstr ""
+
+msgctxt "#32027"
+msgid "Already on newest version for kodi."
```
Comment (MartijnKaijser) [inline resources/language/English/strings.po]: Kodi

File: resources/language/English/strings.po [hunk 8]
```diff
@@ -80,7 +98,21 @@ msgctxt "#32024"
...
+msgstr ""
+
+msgctxt "#32028"
+msgid "No system update available."
+msgstr ""
+
+msgctxt "#32029"
+msgid "Abort during upgrade."
```
Comment (MartijnKaijser) [inline resources/language/English/strings.po]: What does this mean/do?

**Pull request closed without merging**